src/xdist/dsession.py

```diff
@@ -444,26 +444,25 @@ def pytest_xdist_setupnodes(self, specs) -> None:
 
     @pytest.hookimpl
     def pytest_xdist_newgateway(self, gateway) -> None:
-        rinfo = gateway._rinfo()
-        is_local = rinfo.executable == sys.executable
-        if self.config.option.verbose > 0 and not is_local:
-            version = "%s.%s.%s" % rinfo.version_info[:3]
-            self.rewrite(
-                "[%s] %s Python %s cwd: %s"
-                % (gateway.id, rinfo.platform, version, rinfo.cwd),
-                newline=True,
-            )
+        if self.config.option.verbose > 0:
+            rinfo = gateway._rinfo()
+            different_interpreter = rinfo.executable != sys.executable
+            if different_interpreter:
+                version = "%s.%s.%s" % rinfo.version_info[:3]
+                self.rewrite(
+                    f"[{gateway.id}] {rinfo.platform} Python {version} cwd: {rinfo.cwd}",
+                    newline=True,
+                )
         self.setstatus(gateway.spec, WorkerStatus.Initialized, tests_collected=0)
 
     @pytest.hookimpl
     def pytest_testnodeready(self, node) -> None:
-        d = node.workerinfo
-        is_local = d.get("executable") == sys.executable
-        if self.config.option.verbose > 0 and not is_local:
-            infoline = "[{}] Python {}".format(
-                d["id"], d["version"].replace("\n", " -- ")
-            )
-            self.rewrite(infoline, newline=True)
+        if self.config.option.verbose > 0:
```

Comment:
my understandign is that the node_ready is not leaning into the problem, as by then there ought to be the running remote thread with the worker